src/bullsAndCows.js

```diff
@@ -27,6 +27,42 @@
  */
 function bullsAndCows(generatedNumber, enteredNumber) {
   // write code here
+  const storage = {};
+  const regex = /^\d{4}$/;
+  let cows = 0;
+  let bulls = 0;
+
+  if (!(regex.test(enteredNumber))) {
+    return undefined;
+  }
+
+  for (let i = 0; i < generatedNumber.length; i++) {
+    storage[generatedNumber[i]] = i;
+  }
+
+  for (let i = 0; i < enteredNumber.length; i++) {
+    const currentValue = enteredNumber[i];
+    // check for repeating numbers
+
+    if (currentValue === enteredNumber[i + 1]) {
+      return undefined;
+    }
+
```

Comment:
And what if digits repeat not right next to each other, like 1213 or 2432? I'm not sure about this, but according to the rules it has to be 4 **different** digits.

Reply:
Yeap, you right. It was kinda cheating. I will fix that.